src/app/api/user.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core';
+import { BaseApiService, Version } from './base-api.service';
+
+export interface User {
```

Comment:
Why do you export a User interface? Do I need this for meme.service?

Reply:
You can use it for type information since the user object that is returned by the methods here corresponds to this type. You probably don't need to import it unless you are creating user objects.

Reply:
But should I export an interface for Meme data?? o.O

Reply:
It doesn't hurt to do so. You should create an interface for it like above since it'll give type information to the methods that use the objects your service returns.